incompatible to some input method like Japanese or Chinese

**Please use the 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to show that you are affected by the same issue. Please don't comment if you have no relevant information to add!**

**Describe the bug**
when add a checkbox or select options, if i type any char, a new empty input item will appear below.
this behavior will interrupt some input method like Chinese or Japanese, that one character needs multiple type,
and cause error message "There was an issue deleting this option", also the option value is incorrect.

**To Reproduce**
Steps to reproduce the behavior:

1. Click "new form"
2. Click "add a question", select "Dropdown"
3. Click "add a new answer" input field
4. use Japanese input method, type "sa" for char さ

**Expected behavior**
the char "さ" appear in the answer

**Screenshots**
![Screenshot_20240622_155814s](https://github.com/nextcloud/forms/assets/3406567/f3ab0de3-a62d-474c-8a29-6935031cb37e)

**Nextcloud (please complete the following information):**

-   Nextcloud-Version: [Nextcloud Hub 8 (29.0.2)]
-   Forms-Version: [4.2.4]

**Desktop (please complete the following information):**

-   OS: [Windows 11]
-   Browser [chrome]
-   Version [126.0.6478.115]

**Browser log**

```
NcSettingsSection-B0-LmypW-DDIm67GF.mjs:2     DELETE https://nc2.mysite.net/ocs/v2.php/apps/forms/api/v2.4/option/cbrrt 400
(anonymous) @ NcSettingsSection-B0-LmypW-DDIm67GF.mjs:2
xhr @ NcSettingsSection-B0-LmypW-DDIm67GF.mjs:2
qi @ NcSettingsSection-B0-LmypW-DDIm67GF.mjs:2
_request @ NcSettingsSection-B0-LmypW-DDIm67GF.mjs:2
request @ NcSettingsSection-B0-LmypW-DDIm67GF.mjs:2
bs.<computed> @ NcSettingsSection-B0-LmypW-DDIm67GF.mjs:2
(anonymous) @ NcSettingsSection-B0-LmypW-DDIm67GF.mjs:2
deleteOptionFromDatabase @ Submit-DWdudiQx.mjs:2
deleteOption @ Submit-DWdudiQx.mjs:2
(anonymous) @ Submit-DWdudiQx.mjs:2
checkValidOption @ Submit-DWdudiQx.mjs:2
me @ vendor-DFuGXkIP.mjs:2
t @ vendor-DFuGXkIP.mjs:2
me @ vendor-DFuGXkIP.mjs:2
e.$emit @ vendor-DFuGXkIP.mjs:2
handleTabbing @ Submit-DWdudiQx.mjs:2
me @ vendor-DFuGXkIP.mjs:2
t @ vendor-DFuGXkIP.mjs:2
Yc.i._wrapper @ vendor-DFuGXkIP.mjs:2
focus @ Submit-DWdudiQx.mjs:2
focusIndex @ Submit-DWdudiQx.mjs:2
(anonymous) @ Submit-DWdudiQx.mjs:2
(anonymous) @ vendor-DFuGXkIP.mjs:2
Cn @ vendor-DFuGXkIP.mjs:2
Promise.then (async)
an @ vendor-DFuGXkIP.mjs:2
er @ vendor-DFuGXkIP.mjs:2
Ga.e.$nextTick @ vendor-DFuGXkIP.mjs:2
render @ NcSettingsSection-B0-LmypW-DDIm67GF.mjs:2
Ga.e._render @ vendor-DFuGXkIP.mjs:2
r @ vendor-DFuGXkIP.mjs:2
e.get @ vendor-DFuGXkIP.mjs:2
e.run @ vendor-DFuGXkIP.mjs:2
ws @ vendor-DFuGXkIP.mjs:2
(anonymous) @ vendor-DFuGXkIP.mjs:2
Cn @ vendor-DFuGXkIP.mjs:2
Promise.then (async)
an @ vendor-DFuGXkIP.mjs:2
er @ vendor-DFuGXkIP.mjs:2
Rt @ vendor-DFuGXkIP.mjs:2
e.update @ vendor-DFuGXkIP.mjs:2
e.notify @ vendor-DFuGXkIP.mjs:2
set @ vendor-DFuGXkIP.mjs:2
m.<computed> @ vendor-DFuGXkIP.mjs:2
me @ vendor-DFuGXkIP.mjs:2
t @ vendor-DFuGXkIP.mjs:2
me @ vendor-DFuGXkIP.mjs:2
e.$emit @ vendor-DFuGXkIP.mjs:2
updateOptions @ Submit-DWdudiQx.mjs:2
addNewEntry @ Submit-DWdudiQx.mjs:2
me @ vendor-DFuGXkIP.mjs:2
t @ vendor-DFuGXkIP.mjs:2
Yc.i._wrapper @ vendor-DFuGXkIP.mjs:2
NcSettingsSection-B0-LmypW-DDIm67GF.mjs:2 [ERROR] forms: Error while deleting an option {app: 'forms', uid: 'admin', level: 2, option: {…}, error: V}
value @ NcSettingsSection-B0-LmypW-DDIm67GF.mjs:2
value @ NcSettingsSection-B0-LmypW-DDIm67GF.mjs:2
(anonymous) @ Submit-DWdudiQx.mjs:2
Promise.catch (async)
deleteOptionFromDatabase @ Submit-DWdudiQx.mjs:2
deleteOption @ Submit-DWdudiQx.mjs:2
(anonymous) @ Submit-DWdudiQx.mjs:2
checkValidOption @ Submit-DWdudiQx.mjs:2
me @ vendor-DFuGXkIP.mjs:2
t @ vendor-DFuGXkIP.mjs:2
me @ vendor-DFuGXkIP.mjs:2
e.$emit @ vendor-DFuGXkIP.mjs:2
handleTabbing @ Submit-DWdudiQx.mjs:2
me @ vendor-DFuGXkIP.mjs:2
t @ vendor-DFuGXkIP.mjs:2
Yc.i._wrapper @ vendor-DFuGXkIP.mjs:2
focus @ Submit-DWdudiQx.mjs:2
focusIndex @ Submit-DWdudiQx.mjs:2
(anonymous) @ Submit-DWdudiQx.mjs:2
(anonymous) @ vendor-DFuGXkIP.mjs:2
Cn @ vendor-DFuGXkIP.mjs:2
Promise.then (async)
an @ vendor-DFuGXkIP.mjs:2
er @ vendor-DFuGXkIP.mjs:2
Ga.e.$nextTick @ vendor-DFuGXkIP.mjs:2
render @ NcSettingsSection-B0-LmypW-DDIm67GF.mjs:2
Ga.e._render @ vendor-DFuGXkIP.mjs:2
r @ vendor-DFuGXkIP.mjs:2
e.get @ vendor-DFuGXkIP.mjs:2
e.run @ vendor-DFuGXkIP.mjs:2
ws @ vendor-DFuGXkIP.mjs:2
(anonymous) @ vendor-DFuGXkIP.mjs:2
Cn @ vendor-DFuGXkIP.mjs:2
Promise.then (async)
an @ vendor-DFuGXkIP.mjs:2
er @ vendor-DFuGXkIP.mjs:2
Rt @ vendor-DFuGXkIP.mjs:2
e.update @ vendor-DFuGXkIP.mjs:2
e.notify @ vendor-DFuGXkIP.mjs:2
set @ vendor-DFuGXkIP.mjs:2
m.<computed> @ vendor-DFuGXkIP.mjs:2
me @ vendor-DFuGXkIP.mjs:2
t @ vendor-DFuGXkIP.mjs:2
me @ vendor-DFuGXkIP.mjs:2
e.$emit @ vendor-DFuGXkIP.mjs:2
updateOptions @ Submit-DWdudiQx.mjs:2
addNewEntry @ Submit-DWdudiQx.mjs:2
me @ vendor-DFuGXkIP.mjs:2
t @ vendor-DFuGXkIP.mjs:2
Yc.i._wrapper @ vendor-DFuGXkIP.mjs:2
Submit-DWdudiQx.mjs:2 Uncaught (in promise) TypeError: Cannot read properties of undefined (reading 'value')
    at s.onInput (Submit-DWdudiQx.mjs:2:293202)
onInput @ Submit-DWdudiQx.mjs:2
await in onInput (async)
(anonymous) @ Submit-DWdudiQx.mjs:2
(anonymous) @ vendor-DFuGXkIP.mjs:2
Cn @ vendor-DFuGXkIP.mjs:2
Promise.then (async)
an @ vendor-DFuGXkIP.mjs:2
er @ vendor-DFuGXkIP.mjs:2
Ga.e.$nextTick @ vendor-DFuGXkIP.mjs:2
render @ NcSettingsSection-B0-LmypW-DDIm67GF.mjs:2
Ga.e._render @ vendor-DFuGXkIP.mjs:2
r @ vendor-DFuGXkIP.mjs:2
e.get @ vendor-DFuGXkIP.mjs:2
e.run @ vendor-DFuGXkIP.mjs:2
ws @ vendor-DFuGXkIP.mjs:2
(anonymous) @ vendor-DFuGXkIP.mjs:2
Cn @ vendor-DFuGXkIP.mjs:2
Promise.then (async)
an @ vendor-DFuGXkIP.mjs:2
er @ vendor-DFuGXkIP.mjs:2
Rt @ vendor-DFuGXkIP.mjs:2
e.update @ vendor-DFuGXkIP.mjs:2
e.notify @ vendor-DFuGXkIP.mjs:2
set @ vendor-DFuGXkIP.mjs:2
m.<computed> @ vendor-DFuGXkIP.mjs:2
me @ vendor-DFuGXkIP.mjs:2
t @ vendor-DFuGXkIP.mjs:2
me @ vendor-DFuGXkIP.mjs:2
e.$emit @ vendor-DFuGXkIP.mjs:2
updateOptions @ Submit-DWdudiQx.mjs:2
addNewEntry @ Submit-DWdudiQx.mjs:2
me @ vendor-DFuGXkIP.mjs:2
t @ vendor-DFuGXkIP.mjs:2
Yc.i._wrapper @ vendor-DFuGXkIP.mjs:2
```

--- a/src/components/Questions/QuestionDropdown.vue
+++ b/src/components/Questions/QuestionDropdown.vue
@@ -78,14 +78,15 @@
 				<li v-if="!isLastEmpty || hasNoAnswer" class="question__item">
 					<input
 						ref="pseudoInput"
-						v-model="inputValue"
+						class="question__input"
 						:aria-label="t('forms', 'Add a new answer')"
 						:placeholder="t('forms', 'Add a new answer')"
-						class="question__input"
 						:maxlength="maxStringLengths.optionText"
 						minlength="1"
 						type="text"
-						@input="addNewEntry" />
+						@input="addNewEntry"
+						@compositionstart="onCompositionStart"
+						@compositionend="onCompositionEnd" />
 				</li>
 			</ol>
 		</template>
@@ -112,7 +113,6 @@ import IconContentPaste from 'vue-material-design-icons/ContentPaste.vue'
 import AnswerInput from './AnswerInput.vue'
 import OptionInputDialog from '../OptionInputDialog.vue'
 import QuestionMixin from '../../mixins/QuestionMixin.js'
-import GenRandomId from '../../utils/GenRandomId.js'
 import logger from '../../utils/Logger.js'
 
 export default {
@@ -153,7 +153,7 @@ export default {
 
 		isLastEmpty() {
 			const value = this.options[this.options.length - 1]
-			return value?.text?.trim().length === 0
+			return value?.text?.trim?.().length === 0
 		},
 
 		isMultiple() {
@@ -239,37 +239,11 @@ export default {
 		 * @param {object} answer the answer to update
 		 */
 		updateAnswer(id, answer) {
-			const options = this.options.slice()
+			const options = [...this.options]
 			const answerIndex = options.findIndex((option) => option.id === id)
 			options[answerIndex] = answer
 
 			this.updateOptions(options)
-		},
-
-		/**
-		 * Add a new empty answer locally
-		 */
-		addNewEntry() {
-			// Add local entry
-			const options = this.options.slice()
-			options.push({
-				id: GenRandomId(),
-				questionId: this.id,
-				text: this.inputValue,
-				local: true,
-			})
-
-			this.inputValue = ''
-
-			// Update question
-			this.updateOptions(options)
-
-			this.$nextTick(() => {
-				this.focusIndex(options.length - 1)
-
-				// Trigger onInput on new AnswerInput for posting the new option to the API
-				this.$refs.input[options.length - 1].onInput()
-			})
 		},
 
 		/**

--- a/src/components/Questions/QuestionMultiple.vue
+++ b/src/components/Questions/QuestionMultiple.vue
@@ -172,7 +172,9 @@
 							:maxlength="maxStringLengths.optionText"
 							minlength="1"
 							type="text"
-							@input="addNewEntry" />
+							@input="addNewEntry"
+							@compositionstart="onCompositionStart"
+							@compositionend="onCompositionEnd" />
 					</li>
 				</ul>
 			</template>
@@ -205,7 +207,6 @@ import IconRadioboxBlank from 'vue-material-design-icons/RadioboxBlank.vue'
 
 import AnswerInput from './AnswerInput.vue'
 import QuestionMixin from '../../mixins/QuestionMixin.js'
-import GenRandomId from '../../utils/GenRandomId.js'
 import logger from '../../utils/Logger.js'
 import OptionInputDialog from '../OptionInputDialog.vue'
 
@@ -485,10 +486,9 @@ export default {
 		 * So we require the one that are checked or all
 		 * if none are checked yet.
 		 *
-		 * @param {number} id the answer id
 		 * @return {boolean}
 		 */
-		checkRequired(id) {
+		checkRequired() {
 			// false, if question not required
 			if (!this.isRequired) {
 				return false
@@ -551,39 +551,6 @@ export default {
 			options[answerIndex] = answer
 
 			this.updateOptions(options)
-		},
-
-		/**
-		 * Add a new empty answer locally
-		 * @param {InputEvent} event The input event that triggered adding a new entry
-		 */
-		addNewEntry({ target }) {
-			// Add local entry
-			const options = [
-				...this.options,
-				{
-					id: GenRandomId(),
-					questionId: this.id,
-					text: target?.value ?? '',
-					local: true,
-				},
-			]
-
-			// Reset the "new answer" input if needed
-			if (this.$refs.pseudoInput) {
-				this.$refs.pseudoInput.value = ''
-			}
-
-			// Update questions
-			this.updateOptions(options)
-
-			this.$nextTick(() => {
-				// Set focus to the created input element
-				this.focusIndex(options.length - 1)
-
-				// Trigger onInput on new AnswerInput for posting the new option to the API
-				this.$refs.input[options.length - 1].onInput()
-			})
 		},
 
 		/**

--- a/src/mixins/QuestionMixin.js
+++ b/src/mixins/QuestionMixin.js
@@ -26,6 +26,7 @@ import axios from '@nextcloud/axios'
 import debounce from 'debounce'
 
 import logger from '../utils/Logger.js'
+import GenRandomId from '../utils/GenRandomId.js'
 import OcsResponse2Data from '../utils/OcsResponse2Data.js'
 import Question from '../components/Questions/Question.vue'
 
@@ -177,6 +178,14 @@ export default {
 		 * Can question be moved down in order?
 		 */
 		canMoveDown: {
+			type: Boolean,
+			default: false,
+		},
+
+		/**
+		 * isComposing for IME handling
+		 */
+		isIMEComposing: {
 			type: Boolean,
 			default: false,
 		},
@@ -408,6 +417,64 @@ export default {
 				this.focusIndex(options.length - 1)
 			})
 			this.isLoading = false
+		},
+
+		/**
+		 * Add a new empty answer locally
+		 * @param {InputEvent} event The input event that triggered adding a new entry
+		 */
+		addNewEntry({ target, isComposing }) {
+			/*
+			 * Check for !isComposing needed for languages using IME like Japanese or Chinese
+			 * Check for !this.isComposing needed for IME inputs handled by CompositionEvents
+			 * Check for target.value !== '' needed for Linux/Mac for characters like á or è
+			 */
+			if (!isComposing && !this.isIMEComposing && target.value !== '') {
+				// Add local entry
+				const options = [
+					...this.options,
+					{
+						id: GenRandomId(),
+						questionId: this.id,
+						text: target.value,
+						local: true,
+					},
+				]
+
+				// Reset the "new answer" input if needed
+				if (this.$refs.pseudoInput) {
+					this.$refs.pseudoInput.value = ''
+				}
+
+				// Update questions
+				this.updateOptions(options)
+
+				this.$nextTick(() => {
+					// Set focus to the created input element
+					this.focusIndex(options.length - 1)
+
+					// Trigger onInput on new AnswerInput for posting the new option to the API
+					this.$refs.input[options.length - 1].onInput()
+				})
+			}
+		},
+
+		/**
+		 * Handle compostion start event for IME inputs
+		 */
+		onCompositionStart() {
+			this.isIMEComposing = true
+		},
+
+		/**
+		 * Handle compostion end event for IME inputs
+		 * @param {CompositionEvent} event The input event that triggered adding a new entry
+		 */
+		onCompositionEnd({ target, isComposing }) {
+			this.isIMEComposing = false
+			if (!isComposing) {
+				this.addNewEntry({ target, isComposing })
+			}
 		},
 	},
 }

